CoreScriptsRoot/ServerCoreScripts/ClientChat/ChatMain.lua

```diff
@@ -0,0 +1,735 @@
+	if (currentChannelObject) then
+		local Scroller = currentChannelObject.Scroller
+		scrollBarThickness = Scroller.ScrollBarThickness
+		Scroller.ScrollingEnabled = false
+		Scroller.ScrollBarThickness = 0
+	end
+end
+
+local function DoTextFadeIn(setFadingTime)
+	lastFadeTime = tick()
+	textIsFaded = false
+	fadedChanged:Fire()
+	ChatWindow:FadeInText((setFadingTime or defaultFadingTime) * 0)
+end
+
+local function DoTextFadeOut(setFadingTime)
+	lastFadeTime = tick()
+	textIsFaded = true
+	fadedChanged:Fire()
+	ChatWindow:FadeOutText((setFadingTime or defaultFadingTime))
+end
+
+local function DoFadeInFromNewInformation()
+	DoTextFadeIn()
+	DoBackgroundFadeIn()
+end
+
+local function InstantFadeIn()
+	DoBackgroundFadeIn(0)
+	DoTextFadeIn(0)
+end
+
+local function InstantFadeOut()
+	DoBackgroundFadeOut(0)
+	DoTextFadeOut(0)
+end
+
+local function DealWithCoreGuiEnabledChanged(enabled)
+	if (moduleApiTable.Visible) then
+		if (enabled) then
+			InstantFadeIn()
+		else
+			InstantFadeOut()
+		end
+	end
+end
+
+local mouseIsInWindow = nil
+local function UpdateFadingForMouseState(mouseState)
+	mouseIsInWindow = mouseState
+
+	mouseStateChanged:Fire()
+
+	if (ChatBar:IsFocused()) then return end
+
+	if (mouseState) then
+		DoBackgroundFadeIn()
+		DoTextFadeIn()
+	else
+		DoBackgroundFadeIn()
+	end
+end
+
+
+local last = 0
+spawn(function()
+	while true do
+		RunService.RenderStepped:wait()
+
+		while (mouseIsInWindow or ChatBar:IsFocused()) do
+			if (mouseIsInWindow) then
+				mouseStateChanged.Event:wait()
+			end
+			if (ChatBar:IsFocused()) then
+				chatBarFocusChanged.Event:wait()
+			end
+		end
+
+		local timeDiff = tick() - lastFadeTime
+
+		-- debug timer printing
+		--if (math.abs(last - timeDiff) > 0.5) then
+		--	last = timeDiff
+		--	print("Step", timeDiff, backgroundIsFaded, textIsFaded)
+		--end
+
+		if (not backgroundIsFaded) then
+			if (timeDiff > backgroundFadeTimer) then
+				DoBackgroundFadeOut()
+			end
+
+		elseif (not textIsFaded) then
+			if (timeDiff > textFadeTimer) then
+				DoTextFadeOut()
+			end
+
+		else
+			fadedChanged.Event:wait()
+
+		end
+
+	end
+end)
+
+
+local function UpdateMousePosition(mousePos)
+	if (not moduleApiTable.Visible or not moduleApiTable.IsCoreGuiEnabled or not moduleApiTable.TopbarEnabled) then return end
+
+	local windowPos = ChatWindow.GuiObject.AbsolutePosition
+	local windowSize = ChatWindow.GuiObject.AbsoluteSize
+
+	local newMouseState = CheckIfPointIsInSquare(mousePos, windowPos, windowPos + windowSize)
+	if (newMouseState ~= mouseIsInWindow) then
+		UpdateFadingForMouseState(newMouseState)
+	end
+end
+
+UserInputService.InputChanged:connect(function(inputObject)
+	if (inputObject.UserInputType == Enum.UserInputType.MouseMovement) then
+		local mousePos = Vector2.new(inputObject.Position.X, inputObject.Position.Y)
+		UpdateMousePosition(mousePos)
+	end
+end)
+
+UserInputService.TouchTap:connect(function(tapPos, gameProcessedEvent)
```

Comment:
Not sure if this is intentional, but I believe this a high level touch event and only fire on quick tap gestures rather than any touch movement.